src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -147,6 +151,84 @@ const char* GetCalendarName(CalendarId calendarId)
 
 /*
 Function:
+GetCalendarName
+
+Gets the associated CalendarId for the ICU calendar name.
+*/
+CalendarId GetCalendarId(const char* calendarName)
+{
+	if (strcmp(calendarName, "gregorian") == 0)
+		//TODO: what about the other gregorian types?
+		return GREGORIAN;
+	else if (strcmp(calendarName, "japanese") == 0)
+		return JAPAN;
+	else if (strcmp(calendarName, "buddhist") == 0)
+		return THAI;
+	else if (strcmp(calendarName, "hebrew") == 0)
+		return HEBREW;
+	else if (strcmp(calendarName, "chinese") == 0)
+		//TODO: what about these other lunisolar types?
```

Comment:
lunisolar calendars today are not associated with any culture. so such calendars cannot be used for date formatting or parsing. this apply to the Julian calendar too.